client/src/pages/Messages/ActiveChat/Input.tsx

```diff
@@ -67,24 +69,66 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const Input: FC<Props> = (props) => {
-  const { currentUserId, recipientUser, conversationId, dispatchMessages } = props;
+  const { currentUserId, recipientUser, conversationId, dispatchMessages, dispatchConversations } = props;
   const [text, setText] = useState<string>('');
   const classes = useStyles();
+  const { socket } = useSocket();
+
+  useEffect(() => {
+    if (socket === undefined || currentUserId === '') return;
+
+    socket.on('new-message', (data) => {
+      dispatchMessages({ type: 'ADD_NEW_MESSAGE', activeConversation: data.activeConversation, message: data.message });
+      dispatchConversations({
+        type: 'UPDATE_LATEST_MESSAGE',
+        activeConversation: data.activeConversation,
+        message: data.message.text,
+        createdAt: new Date().toISOString(),
+      });

```

Comment:
(optional) These two actions always fire together, should they be combined?

Reply:
Sure, I do that